system/src/system_network_manager.cpp

```diff
@@ -624,6 +645,12 @@ void NetworkManager::handleIfLink(if_t iface, const struct if_event* ev) {
             }
         }
     }
+
+    if (spark_cloud_flag_connected() && disconnectCloud) {
+        auto systemOptions = options.toSystemOptions();
+        spark_cloud_disconnect(&systemOptions, nullptr);
```

Comment:
Should we defer the actual disabling of the `SPARK_CLOUD_AUTO_CONNECT` flag to the system task queue? I'm worried we would get into a situation where the cloud reconnect options would be consumed/overridden before `cloud_disconnect()` runs and re-sets the cloud auto connect flag...